app/policies/task_policy.rb

```diff
@@ -54,7 +54,7 @@ def manage?
   end
 
   def generate_test?
-    Settings.open_ai.access_token.present? and update?
+    user.present? and update?
```

Comment:
I think this should be something like this:
```suggestion
    @user.personal_openai_api_key.present? and update?
```

Reply:
when I did it like that, the errors were not showing properly. It should be possible to click the button regardless of whether you have a key or not it think

Reply:
As described above, I would only let the user click the button, if they have the key. Otherwise disable and tooltip.

Reply:
In the commit mentioned above, I also change the policy, it checks only for the `update?` right now. The API-Key presence check is independent